runbotics/runbotics-orchestrator-ui/src/src-app/components/BotProcessRunner.tsx

```diff
@@ -44,54 +69,53 @@ interface BotProcessRunnerProps {
 const BotProcessRunner: FC<BotProcessRunnerProps> = ({
     className,
     process,
+    externalProcessInstance,
+    isProcessInstanceActive,
     onRunClick,
     color = 'secondary',
     variant = 'text',
 }) => {
     const dispatch = useDispatch();
     const { enqueueSnackbar, closeSnackbar } = useSnackbar();
-    const [started, setStarted] = useState(false);
+    const [started, setStarted] = useState(isProcessInstanceActive);
     const [isSubmitting, setSubmitting] = useState(false);
     const [loading, setLoading] = useState(false);
     const [modalOpen, setModalOpen] = useState(false);
     const { translate } = useTranslations();
     const hasRunProcessAccess = useFeatureKey([FeatureKey.PROCESS_START]);
-    
+
     const processInstances = useSelector(processInstanceSelector);
-    const { activeJobs } = useSelector(schedulerSelector);
     const { orchestratorProcessInstanceId, processInstance } = processInstances.active;
-    const isRunButtonDisabled = started || isSubmitting || !process.system || !process.botCollection;
+    const currentProcessInstance = externalProcessInstance ?? processInstance;
     const isProcessAttended = process?.isAttended && process?.executionInfo;
     const processName = process?.name;
     const processId = process?.id;
-    
-    useEffect(() => {
-        dispatch(schedulerActions.getActiveJobs());
-    }, [dispatch, processId]);
-    
+    const isRunButtonDisabled =
+        started || isSubmitting || !process.system || !process.botCollection;
+    const isRerunButtonDisabled = isProcessActive(processId, processInstance) || started || isSubmitting;
+
     useEffect(() => {
-        if (processId === activeJobs[0]?.process.id) {
+        if (started) return;
+        if (isProcessActive(processId, currentProcessInstance)) {
             setStarted(true);
         }
-    }, [activeJobs, processId]);
-    
-    useEffect(() => {
-        const isProcessInstanceFinished = processInstance?.status === ProcessInstanceStatus.COMPLETED
-            || processInstance?.status === ProcessInstanceStatus.ERRORED
-            || processInstance?.status === ProcessInstanceStatus.TERMINATED;
+    }, [started, processId, currentProcessInstance]);
 
-        if (isProcessInstanceFinished) 
-        { setStarted(false); }
-        
-    }, [processInstance]);
+    useEffect(() => {
+        if (isProcessInstanceFinished(currentProcessInstance)) {
+            setStarted(false);
+        }
+    }, [currentProcessInstance]);
```

Comment:
this one and the useEffect above can be merged, follow the previous hint with redux object

Reply:
I think we can't rely only on the redux object in this case, but it can br merged xd